### PR TITLE
update sa receiver readme

### PR DESCRIPTION
**Pull request by pmcollins**
Updated some stale links in this document. Also, made the language a bit more in line with ELI5 (explain like I'm five), which I happen to like for technical documentation. Would like to do more of these, but if folks don't like this style, please feel free to revise, extend, or revert!

### CODE REVIEW

File: pkg/receiver/smartagent/README.md [hunk 1]
```diff
@@ -2,50 +2,49 @@
 
 > **Note:** The SignalFx Smart Agent receiver is only supported on x86_64/amd64 platforms.
 
-The Smart Agent Receiver allows you to utilize existing [SignalFx Smart Agent monitors](https://github.com/signalfx/signalfx-agent#monitors)
-as OpenTelemetry Collector metric receivers.  It assumes that you have a properly configured environment with a
-functional [Smart Agent release bundle](https://github.com/signalfx/signalfx-agent/releases/latest) on your system,
-which is already provided for x86_64/amd64 Splunk Distribution of OpenTelemetry Collector
```
[the remark this answers is not in the document]
Comment (pmcollins) [reply pkg/receiver/smartagent/README.md:8]: But the note above says `The SignalFx Smart Agent receiver is only supported on x86_64/amd64 platforms.` Should we mention that again here?

[the remark this answers is not in the document]
Comment (rmfitzpatrick) [reply pkg/receiver/smartagent/README.md:8]: sorry, specifically the "installation paths" since the distribution executable doesn't provide them alone.

File: pkg/receiver/smartagent/README.md [hunk 2]
```diff
@@ -2,50 +2,49 @@
...
-1. In lieu of `discoveryRule` support, the Collector's
-[`receivercreator`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/receivercreator/README.md)
-and associated [Observer extensions](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/README.md)
-should be used.
-1. The [`signalfx-forwarder`](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitors/signalfx-forwarder.md)
-monitor should be made part of both `metrics` and `traces` pipelines utilizing the
-[`signalfx`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/signalfxexporter/README.md)
-and [`sapm`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/sapmexporter/README.md) exporters, respectively.
-1. All metric content replacement and transformation rules should utilize existing
+1. Put any Smart Agent or collectd configuration into the
```
Comment (rmfitzpatrick) [inline pkg/receiver/smartagent/README.md:23]: I think stressing "global" is helpful here since "any smart agent or collectd configuration" isn't accurate if users assume an implicit monitor inclusion.

File: pkg/receiver/smartagent/README.md [hunk 3]
```diff
@@ -2,50 +2,49 @@
...
+[Smart Agent Extension section](../../extension/smartagentextension/README.md)
+of your Collector configuration.
+1. Instead of using Smart Agent's `discoveryRule`s, use the Collector's
+[Receiver Creator](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/receivercreator/README.md)
+and [Observer extensions](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/README.md).
+1. Instead of using the [SignalFx Forwarder](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitors/signalfx-forwarder.md)
```
Comment (rmfitzpatrick) [inline pkg/receiver/smartagent/README.md:29]: The signalfx-forwarder monitor in both metrics and trace pipelines was the important bit and it isn't replaced by the   sapm exporter.

**Review by crobert-1 — APPROVED** [review]
I can't speak to correctness, but the style and simplicity works for me.

File: pkg/receiver/smartagent/README.md [hunk 4]
```diff
@@ -2,50 +2,49 @@
...
-`telegraf/*` monitors like `telegraf/exec`.  The `signalfx` exporter is required for sending events to SignalFx, the `splunk_hec` exporter does not support sending events. An example of this is provided below.
+1. If you have a monitor that updates [dimension properties or tags](https://dev.splunk.com/observability/docs/datamodel/metrics_metadata)
+(e.g. `ecs-metadata`, `heroku-metadata`, `kubernetes-cluster`, `openshift-cluster`, `postgresql`, or `sql`), put the name of
+your SignalFx exporter in its `dimensionClients` field in the Collector's SignalFx receiver configuration block.
+If you do not specify any exporters in this array field, the receiver will attempt to use the Collector pipeline to which it's connected. If
+the next element of the pipeline isn't compatible with updating dimensions, and if you configured a single SignalFx exporter,
+the reciever will use that SignalFx exporter. If you don't require dimension updates, you can specify the empty array `[]` to disable it.
+1. If you have a monitor that sends [events](https://dev.splunk.com/observability/docs/datamodel/custom_events) (e.g. `kubernetes-events`,
```
Comment (rmfitzpatrick) [inline pkg/receiver/smartagent/README.md:42]: mind moving this up one in importance since this is required and the dimension client logic has automatic defaults?

File: pkg/receiver/smartagent/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pmcollins) [reply pkg/receiver/smartagent/README.md:8]: Sorry, I'm a bit confused. Are we trying to address people installing a collector executable on a non-x86/amd64 host, which already had a Smart Agent installed on it? Is this supported?

I'm also confused about what is meant by installation paths, and the linked document doesn't mention them. Can you elaborate?

Perhaps if you could rephrase the statement it would help me understand.

File: pkg/receiver/smartagent/README.md (hunk 2 above)
Comment (pmcollins) [reply pkg/receiver/smartagent/README.md]: Sure, done.

File: pkg/receiver/smartagent/README.md (hunk 3 above)
Comment (pmcollins) [reply pkg/receiver/smartagent/README.md]: Nice catch. Thanks.

File: pkg/receiver/smartagent/README.md (hunk 4 above)
Comment (pmcollins) [reply pkg/receiver/smartagent/README.md]: Sure, done.

File: pkg/receiver/smartagent/README.md [hunk 5]
```diff
@@ -2,50 +2,51 @@
...
-and [`sapm`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/sapmexporter/README.md) exporters, respectively.
-1. All metric content replacement and transformation rules should utilize existing
+1. Put any Smart Agent or collectd configuration into the global
+[Smart Agent Extension section](../../extension/smartagentextension/README.md)
+of your Collector configuration.
+1. Instead of using Smart Agent's `discoveryRule`s, use the Collector's
+[Receiver Creator](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/receivercreator/README.md)
+and [Observer extensions](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/README.md).
+1. If using a [SignalFx Forwarder](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitors/signalfx-forwarder.md)
+monitor, put it into both a `metrics` and `traces` pipeline, and use a
```
Comment (rmfitzpatrick) [inline pkg/receiver/smartagent/README.md:30]: nit to respect exporter ordering

```suggestion
monitor, put it into both a `traces` and `metrics` pipeline, and use a
```

File: pkg/receiver/smartagent/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rmfitzpatrick) [reply pkg/receiver/smartagent/README.md:8]: yeah, it's ambiguous overall. I'm not sure how accurate "only supported on x86_64/amd64 platforms" is as a whole since the native go monitors can work for other arches but don't think that'd need to be addressed here.

The SA bundle is only deployed by the x86 installers (and made available to the docker images) and the link to the getting started guide is, I believe, the intro to packaging and installation methods that provide the bundle. ARM or other arches users may be surprised by the bundle not being available to them after installing the rpm, for example, and calling that out here helps provide some expectation setting.

File: pkg/receiver/smartagent/README.md (hunk 5 above)
Comment (pmcollins) [reply pkg/receiver/smartagent/README.md]: Good catch. Fixed.

File: pkg/receiver/smartagent/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pmcollins) [reply pkg/receiver/smartagent/README.md:8]: I see, thanks. I've added some language that hopefully helps folks understand.

[the remark this answers is not in the document]
Comment (rmfitzpatrick) [reply pkg/receiver/smartagent/README.md:8]: thanks, this is clearer than the original.

**Review by rmfitzpatrick — APPROVED** [review]

File: pkg/receiver/smartagent/README.md [hunk 6]
```diff
@@ -1,51 +1,52 @@
...
 
-> **Note:** The SignalFx Smart Agent receiver is only supported on x86_64/amd64 platforms.
+> **Note:** The SignalFx Smart Agent receiver is only fully supported on x86_64/amd64 platforms.
 
-The Smart Agent Receiver allows you to utilize existing [SignalFx Smart Agent monitors](https://github.com/signalfx/signalfx-agent#monitors)
-as OpenTelemetry Collector metric receivers.  It assumes that you have a properly configured environment with a
-functional [Smart Agent release bundle](https://github.com/signalfx/signalfx-agent/releases/latest) on your system,
-which is already provided for x86_64/amd64 Splunk Distribution of OpenTelemetry Collector
-[installation paths](https://github.com/signalfx/splunk-otel-collector#getting-started).
+The Smart Agent Receiver allows you to use [SignalFx Smart Agent monitors](https://github.com/signalfx/signalfx-agent#monitors)
```
Comment (theletterf) [inline pkg/receiver/smartagent/README.md:5]:
```suggestion
The Smart Agent Receiver lets you use [SignalFx Smart Agent monitors](https://github.com/signalfx/signalfx-agent#monitors)
```

File: pkg/receiver/smartagent/README.md [hunk 7]
```diff
@@ -1,51 +1,52 @@
...
+in the [Splunk Distribution of OpenTelemetry Collector](https://github.com/signalfx/splunk-otel-collector). Many
+monitors also require a [Smart Agent release bundle](https://github.com/signalfx/signalfx-agent/releases/latest),
+which is installed by the Splunk Distribution of OpenTelemetry Collector installer on supported x86_64/amd64 platforms.
```
Comment (theletterf) [inline pkg/receiver/smartagent/README.md:8]:
```suggestion
which is installed by the Splunk Distribution of OpenTelemetry Collector on supported x86_64/amd64 platforms.
```

File: pkg/receiver/smartagent/README.md [hunk 8]
```diff
@@ -1,51 +1,52 @@
...
 
-**Beta: All Smart Agent monitors are supported at this time.  Configuration and behavior may change without notice.**
+While this document outlines some of the capabilities of the Smart Agent receiver, please see the
```
Comment (theletterf) [inline pkg/receiver/smartagent/README.md:10]:
```suggestion
See the
```

File: pkg/receiver/smartagent/README.md [hunk 9]
```diff
@@ -1,51 +1,52 @@
...
+[migration guide](../../../docs/signalfx-smart-agent-migration.md)
+for more information about migrating from the Smart Agent to the Splunk Distribution of the OpenTelemetry Collector.
+
+**Beta: All Smart Agent monitors are supported at this time. Configuration and behavior may change without notice.**
```
Comment (theletterf) [inline pkg/receiver/smartagent/README.md:14]:
```suggestion
**Beta: All Smart Agent monitors are supported by Splunk. Configuration and behavior may change without notice.**
```

File: pkg/receiver/smartagent/README.md [hunk 10]
```diff
@@ -1,51 +1,52 @@
...
 
 ## Configuration
 
-Each `smartagent` receiver configuration acts as a drop-in replacement for each supported Smart Agent Monitor
-[configuration](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitor-config.md) with some exceptions:
+For each Smart Agent 
+[monitor](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitor-config.md)
+you want to put into the Collector, add a `smartagent` receiver configuration block. Once configured in the Collector, each
```
Comment (theletterf) [inline pkg/receiver/smartagent/README.md:20]: "add to" instead of "put into"?

File: pkg/receiver/smartagent/README.md [hunk 11]
```diff
@@ -1,51 +1,52 @@
...
+`smartagent` receiver will act as a drop-in replacement for its corresponding Smart Agent monitor.
```
Comment (theletterf) [inline pkg/receiver/smartagent/README.md:21]:
```suggestion
`smartagent` receiver acts as a replacement for its corresponding Smart Agent monitor.
```

File: pkg/receiver/smartagent/README.md [hunk 12]
```diff
@@ -1,51 +1,52 @@
...
-should be used.
-1. The [`signalfx-forwarder`](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitors/signalfx-forwarder.md)
-monitor should be made part of both `metrics` and `traces` pipelines utilizing the
-[`signalfx`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/signalfxexporter/README.md)
-and [`sapm`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/sapmexporter/README.md) exporters, respectively.
-1. All metric content replacement and transformation rules should utilize existing
+1. Put any Smart Agent or collectd configuration into the global
+[Smart Agent Extension section](../../extension/smartagentextension/README.md)
+of your Collector configuration.
+1. Instead of using Smart Agent's `discoveryRule`s, use the Collector's
```
Comment (theletterf) [inline pkg/receiver/smartagent/README.md:26]:
```suggestion
1. Instead of using `discoveryRule`, use the Collector's
```

File: pkg/receiver/smartagent/README.md [hunk 13]
```diff
@@ -1,51 +1,52 @@
...
+[Receiver Creator](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/receivercreator/README.md)
+and [Observer extensions](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/README.md).
+1. If using a [SignalFx Forwarder](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitors/signalfx-forwarder.md)
```
Comment (theletterf) [inline pkg/receiver/smartagent/README.md:29]:
```suggestion
1. If you're using a [SignalFx Forwarder](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitors/signalfx-forwarder.md)
```

File: pkg/receiver/smartagent/README.md [hunk 14]
```diff
@@ -1,51 +1,52 @@
...
-1. In lieu of `discoveryRule` support, the Collector's
-[`receivercreator`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/receivercreator/README.md)
-and associated [Observer extensions](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/README.md)
-should be used.
-1. The [`signalfx-forwarder`](https://github.com/signalfx/signalfx-agent/blob/main/docs/monitors/signalfx-forwarder.md)
-monitor should be made part of both `metrics` and `traces` pipelines utilizing the
-[`signalfx`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/signalfxexporter/README.md)
-and [`sapm`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/sapmexporter/README.md) exporters, respectively.
-1. All metric content replacement and transformation rules should utilize existing
+1. Put any Smart Agent or collectd configuration into the global
```
Comment (theletterf) [inline pkg/receiver/smartagent/README.md:23]: This is a rather long list of options. Consider splitting into subsections and add an example for each (or link to one)

File: pkg/receiver/smartagent/README.md [hunk 15]
```diff
@@ -1,51 +1,52 @@
...
-to ensure that host identity and other useful information is made available as event dimensions.
-Receiver entries that should be added to logs pipelines include `kubernetes-events`, `nagios`, `processlist`, and potentially any
-`telegraf/*` monitors like `telegraf/exec`.  The `signalfx` exporter is required for sending events to SignalFx, the `splunk_hec` exporter does not support sending events. An example of this is provided below.
+1. If you have a monitor that sends [events](https://dev.splunk.com/observability/docs/datamodel/custom_events) (e.g. `kubernetes-events`,
+`nagios`, `processlist`, and some `telegraf` monitors like `telegraf/exec`), put it in a `logs` pipeline that uses a
+[SignalFx exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/signalfxexporter/README.md).
+It's recommended, and in the case of the Processlist monitor required, to put into the same pipeline a
+[Resource Detection processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md),
+which will add host information and other useful dimensions to the events. An example is provided below.
```
Comment (theletterf) [inline pkg/receiver/smartagent/README.md:42]:
```suggestion
1. If you have a monitor that sends [events](https://dev.splunk.com/observability/docs/datamodel/custom_events), for example `kubernetes-events`,
`nagios`, `processlist`, and some `telegraf` monitors like `telegraf/exec`, put it in a `logs` pipeline that uses a
[SignalFx exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/signalfxexporter/README.md).
If the Processlist monitor is required, add to the same pipeline a
[Resource Detection processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md),
which adds host information and other useful dimensions to the events. See the example.
```

I suggest linking to the example. Positional language ("below") isn't very useful in this case.

File: pkg/receiver/smartagent/README.md [hunk 16]
```diff
@@ -1,51 +1,52 @@
...
+1. If you have a monitor that updates [dimension properties or tags](https://dev.splunk.com/observability/docs/datamodel/metrics_metadata)
+(e.g. `ecs-metadata`, `heroku-metadata`, `kubernetes-cluster`, `openshift-cluster`, `postgresql`, or `sql`), put the name of
+your SignalFx exporter in its `dimensionClients` field in the Collector's SignalFx receiver configuration block.
+If you do not specify any exporters in this array field, the receiver will attempt to use the Collector pipeline to which it's connected. If
+the next element of the pipeline isn't compatible with updating dimensions, and if you configured a single SignalFx exporter,
+the reciever will use that SignalFx exporter. If you don't require dimension updates, you can specify the empty array `[]` to disable it.
```
Comment (theletterf) [inline pkg/receiver/smartagent/README.md:48]:
```suggestion
1. If you have a monitor that updates [dimension properties or tags](https://dev.splunk.com/observability/docs/datamodel/metrics_metadata), for example `ecs-metadata`, `heroku-metadata`, `kubernetes-cluster`, `openshift-cluster`, `postgresql`, or `sql`, put the name of
your SignalFx exporter in its `dimensionClients` field in the Collector's SignalFx receiver configuration block.
If you don't specify any exporters in this array field, the receiver attempts to use the Collector pipeline to which it's connected. If
the next element of the pipeline isn't compatible with updating dimensions, and if you configured a single SignalFx exporter,
the receiver uses that SignalFx exporter. If you don't require dimension updates, you can specify the empty array `[]` to disable it.
```

File: pkg/receiver/smartagent/README.md (hunk 10 above)
Comment (pmcollins) [reply pkg/receiver/smartagent/README.md]: Better. Fixed throughout.

File: pkg/receiver/smartagent/README.md (hunk 11 above)
Comment (pmcollins) [reply pkg/receiver/smartagent/README.md]: fixed

File: pkg/receiver/smartagent/README.md (hunk 14 above)
Comment (pmcollins) [reply pkg/receiver/smartagent/README.md:23]: Thanks, mind if we hand this off to your team?

File: pkg/receiver/smartagent/README.md (hunk 15 above)
Comment (pmcollins) [reply pkg/receiver/smartagent/README.md]: Mind if we hand this off to your team?
